rq/worker.py

```diff
@@ -610,8 +610,13 @@ def monitor_work_horse(self, job):
         """
         while True:
             try:
-                self._monitor_work_horse_tick(job)
+                with UnixSignalDeathPenalty(DEFAULT_JOB_MONITORING_INTERVAL, HorseMonitorTimeoutException):
+                    retpid, ret_val = os.waitpid(self._horse_pid, 0)
                 break
+            except HorseMonitorTimeoutException:
+                # Horse has not exited yet and is still running.
+                # Send a heartbeat to keep the worker alive.
+                self.heartbeat(DEFAULT_JOB_MONITORING_INTERVAL + 15)
```

Comment:
Nitpick, I think we can safely change this to `DEFAULT_JOB_MONITORING_INTERVAL + 5` so dead workers will be discovered faster.